### DISCUSSION TITLE
关于1p目前的细节问题建议(2024.3.9)

### DISCUSSION THREAD

**Discussion by defeatedperson** [1Panel-dev/1Panel#4141] [Ideas]
### 1Panel 版本

1.10

### 请描述您的需求或者改进建议

目前，
我了解到的关于1p不好的情况:(普通用户当中)
细节方面不是很友好:
例如
1.文件目录(应该算底层问题)
普通人可能觉得有点乱，
基本只能找入口点进去，一层一层往下翻比较困难
(例如:找站点文件夹，一层一层翻需要很久，不是很直观)

2.常用的小功能不是很方便
例如ftp，部分软件的安装不知道怎么操作-没有提示，php加密拓展(手动安装)不便

3.对内存消耗有点大，
尤其是有独立管理页面的软件
(这个算没办法，但还是尽量优化)

4.对nat服务器(或者集群)不友好
这里指的是安装应用的时候，被waf拦截(原因:请求次数太多)导致无法安装(无法获取应用)

5.站点设置-跳转站点目录按钮
我了解到不少用户，进去就直接把index/waf/ssl文件夹全删了，让后把源码放进站点文件夹(/站点域名)

### 请描述你建议的实现方案

1.可以考虑慢慢优化一下文件夹的逻辑，挂载的目录直观一点(环境文件夹/缓存文件夹/站点运行文件夹/日志文件夹/备份文件夹)

2.例如ftp之类的功能，建议是面板本身就集成管理功能，让用户手动安装即可调用(就和uwaf一样的做法)
对于php的设置，可以考虑在环境配置页面，加入【跳转1p论坛/php板块】的按钮，让用户折腾起来，更方便的看到教程

3.同2，部分常用的小工具，就建议:让用户直装+面板调用的方式

4.可以考虑在应用商店加上人机验证模块(输入图片验证码/滑块验证/点击验证，均可)，对于触发waf拦截的服务器，弹出人机验证，操作者通过之后，给下载waf发信息，把当前IP解除10分钟并观察，若10分钟内无异常(请求次数过多)，就解除观察

5.最简单的方法，将这个按钮，点击就默认打开index文件夹，而不是整个站点目录(域名/*)

### 附加信息

我只是一个整天折腾服务器的普通人，善于折腾细节+对小白友好的改进方案，希望对1panel有帮助

**Comment by mobeicanyue** [comment]
1. 站点文件夹在网页那里可以直接跳转的，有个文件夹的logo

**Reply by defeatedperson** [reply]
我只是举个例子，知道有跳转按钮哈！
目前1p的文件夹，有点不清晰(对于普通用户)，有点门槛

**Reply by wanghe-fit2cloud** [reply]
v2.0 版本重构时，会重点优化当前版本中存在的网站目录层级过深问题。

**Comment by Luxcis** [comment]
计划任务中定时备份快照上传至第三方存储容器中，并设置快照保留3份，第4天执行时不会将第三方容器中的快照删除，导致时间越久快照占用的空间越多最终容量不足导致快照备份失败

**Reply by ssongliu** [reply]
用的啥第三方存储，麻烦提供一下系统日志
